mw-webapp/src/service/GoalMetricService.ts

```diff
@@ -57,9 +57,9 @@ export class GoalMetricService {
   /**
    * Create GoalMetricsDTO with Batch
    */
-  public static async createGoalMetricsDTOWithBatch(
+  public static createGoalMetricsDTOWithBatch(
     goalMetricDTOWithoutUuid: GoalMetricDTOWithoutUuid,
-    batch: WriteBatch): Promise<GoalMetricDTO> {
+    batch: WriteBatch): GoalMetricDTO {
```

Comment:
Interesting how does it work. How can you get. New uuid synchronously for example?

Reply:
You call batch asynchronously in DAL layer